packages/ai-jsx/src/batteries/sidekick/platform/large-response-handler.tsx

```diff
@@ -0,0 +1,228 @@
+import * as AI from '../../../index.js';
+import { ConversationMessage, FunctionResponse, FunctionResponseProps } from '../../../core/conversation.js';
+
+import { RecursiveCharacterTextSplitter } from 'langchain/text_splitter';
+import { getEncoding } from 'js-tiktoken';
+import yaml from 'js-yaml';
+import _ from 'lodash';
+import { UseToolsProps } from '../../use-tools.js';
+import { cohereContext, MarkdownChunkFormatter, RerankerFormatted } from '../../../lib/cohere.js';
+
+export type LengthFunction = ((text: string) => number) | ((text: string) => Promise<number>);
```

Comment:
```suggestion
export type LengthFunction = (text: string) => Promisable<number>
```

https://github.com/sindresorhus/type-fest/blob/main/source/promisable.d.ts

Reply:
But LC uses this signature and TypeScript doesn't understand that they are the same 😢

Reply:
Bummer. Ok, that's fine.

What if we tried this then?

```suggestion
export type LengthFunction = ((text: string) => number | Promise<number>);
```

Reply:
Same issue.

I might end up ejecting (i.e. copy&pasting) the text splitter from LC. If I do that I'll make sure to use `Promisable` instead.